### PR TITLE
Started to work on fixing the serialization code

### PR DESCRIPTION
**Pull request by hkaiser**
This fixes a couple of potential serialization problems, this also makes sure types are marked as bit-wise serializable, where appropriate.

Note: this is best used in conjunction with https://github.com/STEllAR-GROUP/hpx/pull/2511

### CODE REVIEW

File: src/grid.cpp [hunk 1]
```diff
@@ -1507,18 +1510,17 @@ void grid::reconstruct() {
...
 		auto minmod_step = [](real& lhs, real const& r1, real const& r2, real const& r3)
 		{
 			lhs = minmod(lhs, 2.0 * minmod(r1 - r2, r2 - r3));
 		};
 
 		for (integer iii = H_NX * H_NX; iii != H_N3 - H_NX * H_NX; ++iii) {
-			average(slpx[sy_i][iii], slpy[sx_i][iii]);
-			average(slpx[sz_i][iii], slpz[sx_i][iii]);
-			average(slpy[sz_i][iii], slpz[sy_i][iii]);
+			inplace_average(slpx[sy_i][iii], slpy[sx_i][iii]);
```
Comment (brycelelbach) [inline src/grid.cpp]: What is inplace_average?

File: src/grid.cpp [hunk 2]
```diff
@@ -1456,9 +1459,9 @@ inline void limit_slope(real& ql, real q0, real& qr) {
 //    if (qr < q0 || q0 < ql) {
 	if (bool(qr < q0) != bool(q0 < ql)) {
 		qr = ql = q0;
-	} else if (tmp1 * (q0 - 0.5 * tmp2) > sqr(tmp1) / 6.0) {
+	} else if (tmp1 * (q0 - 0.5 * tmp2) > sqr(tmp1) * SIXTH) {
 		ql = 3.0 * q0 - 2.0 * qr;
-	} else if (-(sqr(tmp1) / 6.0) > tmp1 * (q0 - 0.5 * tmp2)) {
+	} else if (-(sqr(tmp1) * SIXTH) > tmp1 * (q0 - 0.5 * tmp2)) {
```
Comment (brycelelbach) [inline src/grid.cpp]: Did you check if this needed? It's fine, but usually the compiler should catch this. If it's not, we should keep an eye out for this and make sure we lift constant divides wherever possible.

File: src/grid.cpp [hunk 3]
```diff
@@ -16,6 +16,9 @@
 #include <cassert>
 
 #include <hpx/include/runtime.hpp>
+#if defined(HPX_HAVE_DATAPAR)
+#include <hpx/include/parallel_transform.hpp>
```
Comment (brycelelbach) [inline src/grid.cpp]: I don't see transform introduced by this commit?

File: src/grid.cpp (hunk 1 above)
Comment (brycelelbach) [inline src/grid.cpp]: Nevermind, found where it's defined. Looks good.

File: src/grid.cpp (hunk 2 above)
Comment (hkaiser) [inline src/grid.cpp]: I have seen performance improvements for this kind of changes for MSVC, not sure whether this has any effect on other compilers.

File: src/grid.cpp (hunk 3 above)
Comment (hkaiser) [inline src/grid.cpp]: Uhh, a left over. Thanks for catching this.

**Pull request merged**